src/main/java/org/opensrp/web/utils/ClientFormValidator.java

```diff
@@ -112,6 +119,27 @@ private void initialisePropertiesFileJsonPathReferences() {
         return missingPropertyFileReferences;
     }
 
+    public HashSet<String> checkForMissingYamlPropertyFileReferences(@NonNull String fileContent) {
+        HashSet<String> propertyFileReferences = new HashSet<>();
+        HashSet<String> missingPropertyFileReferences = new HashSet<>();
+        try {
+            Map<Object, Object> document = new Yaml().load(fileContent);
+            if (document.containsKey(PROPERTIES_FILE_NAME)) {
+                propertyFileReferences.add((String) document.get(PROPERTIES_FILE_NAME));
+            }
+            // Check if the references exist in the DB
+            for (String propertyFileReference: propertyFileReferences) {
+                if (!clientFormService.isClientFormExists(propertyFileReference) && !clientFormService.isClientFormExists(propertyFileReference + ".properties")) {
+                    missingPropertyFileReferences.add(propertyFileReference);
+                }
+            }
+        }catch (ComposerException exception) {
+            logger.error("Validator parsing a YAML that doesn't conform", exception);
```

Comment:
I think you can let the server crash and return the error in a readable and usable format instead of silently handling it. This would mean that for some reason, the yaml passed the sanity check but it still does not conform to the standards

Reply:
Based on the discussion, it seems this is expected and enables us to avoid processing `rules/calculation` files with a yaml structure starting with `---`

Reply:
Yeah, also we've discovered it isn't possible to upload YAML files that reference properties files because the initial check when uploading expects YAML files that have conditions to be processed by the rules engine. This isn't the case with the second set of files. 
I'll add an additional check in `checkValidJsonYamlPropertiesStructure()` to check for the non rules/calculations YAML files structure & allow for such uploads if they're valid